powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/TemporaryKeyBehavior.java

```diff
@@ -0,0 +1,371 @@
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.crypto.SecretKey;
+import java.security.InvalidKeyException;
+import java.security.KeyPair;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.interfaces.ECPrivateKey;
+import java.security.interfaces.ECPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.time.Instant;
+import java.util.*;
+
+/**
+ * Behavior class implementing the temporary key request related processes.
+ *
+ * @author Petr Dvorak, petr@wultra.com
+ */
+@Service
+@Slf4j
+public class TemporaryKeyBehavior {
+
+    @Data
+    private static class TemporaryKeyResult {
+        private byte[] secretKeyBytes;
+        private PrivateKey privateKey;
+        private PublicKey publicKey;
+    }
+
+    private final PowerAuthServiceConfiguration powerAuthServiceConfiguration;
+
+    private final LocalizationProvider localizationProvider;
+    private final ApplicationVersionRepository applicationVersionRepository;
+    private final ActivationRepository activationRepository;
+    private final MasterKeyPairRepository masterKeyPairRepository;
+    private final TemporaryKeyRepository temporaryKeyRepository;
+    private final TemporaryPrivateKeyConverter temporaryPrivateKeyConverter;
+    private final ServerPrivateKeyConverter serverPrivateKeyConverter;
+
+    private final KeyConvertor keyConvertor = new KeyConvertor();
+    private final KeyGenerator keyGenerator = new KeyGenerator();
+    private final PowerAuthServerKeyFactory keyFactory = new PowerAuthServerKeyFactory();
+
+    @Autowired
+    public TemporaryKeyBehavior(PowerAuthServiceConfiguration powerAuthServiceConfiguration, LocalizationProvider localizationProvider, ApplicationVersionRepository applicationVersionRepository, ActivationRepository activationRepository, MasterKeyPairRepository masterKeyPairRepository, TemporaryKeyRepository temporaryKeyRepository, TemporaryPrivateKeyConverter temporaryPrivateKeyConverter, ServerPrivateKeyConverter serverPrivateKeyConverter) {
+        this.powerAuthServiceConfiguration = powerAuthServiceConfiguration;
+        this.localizationProvider = localizationProvider;
+        this.applicationVersionRepository = applicationVersionRepository;
+        this.activationRepository = activationRepository;
+        this.masterKeyPairRepository = masterKeyPairRepository;
+        this.temporaryKeyRepository = temporaryKeyRepository;
+        this.temporaryPrivateKeyConverter = temporaryPrivateKeyConverter;
+        this.serverPrivateKeyConverter = serverPrivateKeyConverter;
+    }
+
+    @Transactional
+    public TemporaryPublicKeyResponse requestTemporaryKey(TemporaryPublicKeyRequest request) throws GenericServiceException, InvalidKeySpecException, CryptoProviderException, GenericCryptoException, InvalidKeyException {
+
+        // Get claims from JWT
+        final DecodedJWT decodedJWT = JWT.decode(request.getJwt());
```

Comment:
There's unhandled exception when `jwt` parameter in the request is `null` leading to 500 server response.

Reply:
Added a null check...